### PR TITLE
Expose SearchSpace through the C API

### PR DESCRIPTION
**Pull request by Elarnon**
This add additional wrappers to the C API for creating a SearchSpace
from an IR function, as well as `telamon_explore` allowing to run the
search from the C API. There are still some limitations, notably:

 - Action objects can be constructed, but the constants for creating
   enum values (DimKind, etc.) are not exposed yet because cbindgen
   does not parse associated constants.

 - The exploration is currently "all or nothing", i.e. there is no way
   to exerce more precise control on what happens in the exploration or
   drive the exploration from the C side.

Those will be addressed in follow-up PRs.

### CODE REVIEW

File: src/ir/instruction.rs [hunk 1]
```diff
@@ -7,6 +7,7 @@ use utils::*;
 /// Uniquely identifies an instruction.
 #[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
 #[repr(C)]
+/// cbindgen:field-names=[id]
```
Comment (ulysseB) [inline src/ir/instruction.rs]: Is this kind of statement documented somewhere ?

File: telamon-capi/src/explorer.rs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+/// The resulting config object must be freed using
+/// `telamon_config_free`.
+#[no_mangle]
+pub extern "C" fn telamon_config_new() -> *mut Config {
+    Box::into_raw(Box::new(Config::default()))
+}
+
+/// Frees an explorer configuration.
+#[no_mangle]
+pub unsafe extern "C" fn telamon_config_free(config: *mut Config) -> TelamonStatus {
```
Comment (ulysseB) [inline telamon-capi/src/explorer.rs]: Why return a status ?

File: src/ir/instruction.rs (hunk 1 above)
Comment (Elarnon) [reply src/ir/instruction.rs]: In cbindgen source code... I should make a PR to document them someday.

File: telamon-capi/src/explorer.rs (hunk 2 above)
Comment (Elarnon) [reply telamon-capi/src/explorer.rs]: Because at some point I was failing if `config` was null. I'll remove it.

**Comment by Elarnon** [comment]
bors r=ulysseB

**Pull request merged**